compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -720,22 +720,25 @@ impl<'a> AstValidator<'a> {
 fn validate_generic_param_order<'a>(
     sess: &Session,
     handler: &rustc_errors::Handler,
-    generics: impl Iterator<
-        Item = (
-            &'a GenericParamKind,
-            ParamKindOrd,
-            Option<&'a [GenericBound]>,
-            Span,
-            Option<String>,
-        ),
-    >,
+    generics: impl Iterator<Item = &'a GenericParam>,
```

Comment:
We could just pass a slice of `GenericParam`s instead, which would be slightly simpler.